src/AWS.Deploy.Recipes/CdkTemplates/AspNetAppEcsFargate/Generated/RecipeConstruct.cs

```diff
@@ -0,0 +1,342 @@
+using System;
+using System.Collections.Generic;
+
+using Amazon.CDK;
+using Amazon.CDK.AWS.ApplicationAutoScaling;
+using Amazon.CDK.AWS.EC2;
+using Amazon.CDK.AWS.ECS;
+using Amazon.CDK.AWS.ElasticLoadBalancingV2;
+using Amazon.CDK.AWS.IAM;
+using AWS.Deploy.Recipes.CDK.Common;
+using AspNetAppEcsFargate.Configurations;
+using Amazon.CDK.AWS.ECR;
+
+
+// This is a generated file from the original deployment recipe. It is recommended to not modify this file in order
+// to allow easy updates to the file when the original recipe that this project was created from has updates.
+// To customize the CDK constructs created in this file you should use the AppStack.CustomizeCDKProps() method.
+
+namespace AspNetAppEcsFargate
+{
+    using static AWS.Deploy.Recipes.CDK.Common.CDKRecipeCustomizer<RecipeConstruct>;
+
+
+    public class RecipeConstruct : Construct
+    {
+        public IVpc? AppVpc { get; private set; }
+        public ICluster? EcsCluster { get; private set; }
+        public IRole? AppIAMTaskRole { get; private set; }
+        public TaskDefinition? AppTaskDefinition { get; private set; }
+        public IRepository? EcrRepository { get; private set; }
+        public ContainerDefinition? AppContainerDefinition { get; private set; }
+        public SecurityGroup? WebAccessSecurityGroup { get; private set; }
+        public IList<ISecurityGroup>? EcsServiceSecurityGroups { get; private set; }
+        public FargateService? AppFargateService { get; private set; }
+        public IApplicationLoadBalancer? ServiceLoadBalancer { get; private set; }
+        public IApplicationListener? LoadBalancerListener { get; private set; }
+        public ApplicationTargetGroup? ServiceTargetGroup { get; private set; }
+        public AwsLogDriver? AppLogging { get; private set; }
+
+        public ScalableTaskCount? AutoScalingConfiguration { get; private set; }
+
+        public string AutoScaleTaskCount { get; } = "AutoScaleTaskCount";
+        public string AutoScaleTypeCpuType { get; } = "CpuScaling";
+        public string AutoScaleTypeRequestType { get; } = "RequestScaling";
+        public string AutoScaleTypeMemoryType { get; } = "MemoryScaling";
+
+        public string AddTargetGroup { get; } = "AddTargetGroup";
```

Comment:
Can these be constants? 

Also, may be worth putting inside a (inner) class.

Reply:
Normally when customizing the props for a CDK construct the user would write code that looks like 
```
if (string.Equals(evnt.ResourceLogicalName, nameof(evnt.Construct.AppContainerDefinition)))
{
   // Customize props
}
```

So to avoid having magical strings in the user's `CustomizeCDKProps` method I suggest the user does a `nameof` operation on the `evnt.Construct.<CDK-Constuct-Property>`.

These CDK properties are used in methods that don't return an object so I can't use the exact same pattern. But by having these as instance properties the end user experience is exact same of calling `evnt.Construct.AutoScaleTypeCpuType`. If I make these constants then the user has to have different behavior for these CDK properties and access them as static from the type name. I don't want users to have to know about the difference.

I did delete `AutoScaleTaskCount` as that was not needed since there was the `AutoScalingConfiguration` property. I also made sure the values for each of the properties match the property name to make sure the `nameof` operator worked correctly.